zerocopy-derive/src/lib.rs

```diff
@@ -634,6 +635,47 @@ fn derive_from_zeros_struct(ast: &DeriveInput, strct: &DataStruct) -> proc_macro
 // - all of its variants are fieldless
 // - one of the variants has a discriminant of `0`
 
+fn find_zero_variant<'a>(variants: impl IntoIterator<Item = &'a Variant>) -> Option<usize> {
```

Comment:
How does this interact with reprs?
- Are there any reprs this is invalid for?
- Does it handle discriminants that don't fit in the given repr? E.g., `#[repr(u8)]` combined with `A = 300`

Reply:
Reworked this to handle signed and unsigned discriminants of all sizes.